### PR TITLE
Bug 1832037 - Migrate HomeScreen.java code to Kotlin.

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1832037

### CODE REVIEW

File: focus-android/app/src/main/java/org/mozilla/focus/shortcut/HomeScreen.kt [hunk 1]
```diff
@@ -0,0 +1,122 @@
+/* -*- Mode: Java; c-basic-offset: 4; tab-width: 4; indent-tabs-mode: nil; -*-
```
Comment (t-p-white) [inline focus-android/app/src/main/java/org/mozilla/focus/shortcut/HomeScreen.kt:1]:
```suggestion
/* 
```

File: focus-android/app/src/main/java/org/mozilla/focus/shortcut/HomeScreen.kt [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
```
Comment (t-p-white) [inline focus-android/app/src/main/java/org/mozilla/focus/shortcut/HomeScreen.kt:4]:
```suggestion
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
```

File: focus-android/app/src/main/java/org/mozilla/focus/shortcut/HomeScreen.kt [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        shortcutIntent.putExtra(REQUEST_DESKTOP, requestDesktop)
+        shortcutIntent.putExtra(ADD_TO_HOMESCREEN_TAG, ADD_TO_HOMESCREEN_TAG)
+        return shortcutIntent
+    }
+
+    /**
+     * Generates a new default title based on the URL.
+     */
+    @VisibleForTesting
+    fun generateTitleFromUrl(url: String?): String {
```
Comment (t-p-white) [inline focus-android/app/src/main/java/org/mozilla/focus/shortcut/HomeScreen.kt:108]: I don't think we should allow a nullable String here, wdyt? Should we also handle the case when 'host' is null

Comment (mcarare) [reply focus-android/app/src/main/java/org/mozilla/focus/shortcut/HomeScreen.kt]: Good call! A null string would throw NPE.

File: focus-android/app/src/test/java/org/mozilla/focus/shortcut/HomeScreenTest.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import org.junit.runner.RunWith
+import org.mozilla.focus.shortcut.HomeScreen.generateTitleFromUrl
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class HomeScreenTest {
+    @Test
+    fun testGenerateTitleFromUrl() {
+        Assert.assertEquals("mozilla.org", generateTitleFromUrl("https://www.mozilla.org"))
+        Assert.assertEquals("facebook.com", generateTitleFromUrl("http://m.facebook.com/home"))
```
Comment (t-p-white) [inline focus-android/app/src/test/java/org/mozilla/focus/shortcut/HomeScreenTest.kt:17]: could add a test here for URL without a host too

**Pull request merged**